src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,151 @@
+    //TODO: convert to lowercase in processing (figure out acronyms)
+    List<String> titleWords = SearchServlet.getSeparateWords(title);
+    
+    //process title words
+    Map<String, Integer> titleMap = new HashMap<String, Integer>();
+    for (String w : titleWords) {
+      w = w.toLowerCase();
+      if (titleMap.containsKey(w)) {
+        titleMap.put(w, titleMap.get(w) + 1);
+      } else {
+        titleMap.put(w, 1);
+      }
+    }
+    List<String> descWords = SearchServlet.getSeparateWords(desc);
+    //process desc words
+    Map<String, Integer> descMap = new HashMap<String, Integer>();
+    for (String w : descWords) {
+      w = w.toLowerCase();
+      if (descMap.containsKey(w)) {
+        descMap.put(w, descMap.get(w) + 1);
+      } else {
+        descMap.put(w, 1);
+      }
+    }
+
+    List<Map.Entry<String, Integer>> titleMapList = new ArrayList<Map.Entry<String, Integer>>(titleMap.entrySet()); 
+
+    //merge lists
+    List<Map.Entry<String, Integer>> mergeList = new ArrayList<Map.Entry<String, Integer>>(descMap.entrySet());
+    for (Map.Entry titleEntry : titleMapList) {
+      boolean found = false;
+      for (int i = 0; i < mergeList.size(); i++) {
+        Map.Entry mergeEntry = mergeList.get(i);
+        if (mergeEntry.getKey().toString().equals(titleEntry.getKey().toString())) {
+          mergeEntry.setValue(((int) mergeEntry.getValue()) + ((int) titleEntry.getValue() * 2));
+          found = true;
+        }
+      }
+      if (!found) {
+        Map.Entry<String,Integer> entry = 
+            new AbstractMap.SimpleEntry<String, Integer>(
+            titleEntry.getKey().toString(), ((int) titleEntry.getValue()) * 2 );
+        mergeList.add(entry);
+      }
+    }
+
+    Collections.sort(mergeList, new Comparator<Map.Entry<String, Integer>>() { 
+      public int compare(Map.Entry<String, Integer> o1,  
+                         Map.Entry<String, Integer> o2) { 
+        return (o2.getValue()).compareTo(o1.getValue()); 
+      } 
+    });
+
+    //add top results to the final list
+    List<String> finalList = new ArrayList<String>();
+    int count = 0;
+    //minimum number of instances to be considered a keyword
+    int cutoff = 2;
+    while (finalList.size() < 5) {
+      Map.Entry e = mergeList.get(count);
+      //exclude words with less appearances than the cutoff
+      if (((int) e.getValue()) < cutoff) break;
+      else if (count >= mergeList.size()) break;
+      //exclude common useless words (in, a, the, etc)
+      else if (!irrelevantWords.contains(e.getKey().toString())) {
+        finalList.add(e.getKey().toString());
+      }
+      count++;
+    }
+
+    return finalList;
+  }
+
+  /**
+   * Returns a list of words contained in a string
+   */
+  public static List<String> getSeparateWords(String str) {
+    // TODO: handle single quoted strings (right now keeps them in for
+    // the common case of contractions
+
+    String[] list = str.split("([\\s.,!?:;()\\[\\]&\"\\s+])");
```

Comment:
Would be good to add a comment here to help the reader understand the code

Reply:
Done.